src/scene/graph-node.js

```diff
@@ -594,11 +593,22 @@ class GraphNode extends EventHandler {
         const results = [];
         const test = createTest(attr, value);
 
-        this.forEach((node) => {
+        const stack = GraphNode._tmpGraphNodeStack;
```

Comment:
Since `forEach` is now optimized to execute iteratively, why duplicate the iterative loop again here?